### ISSUE TITLE
Error when compiling for 32-bit Linux

### ISSUE THREAD

**Issue by loziniak** [maidsafe/qp2p#244]
**Describe the bug**
I get an error, when compiling for 32-bit Linux:

```
   Compiling qp2p v0.10.1
error[E0308]: mismatched types
   --> /home/devel/.cargo/registry/src/github.com-1ecc6299db9ec823/qp2p-0.10.1/src/wire_msg.rs:156:45
    |
156 |         let data_len = usize::from_be_bytes([0, 0, 0, 0, bytes[2], bytes[3], bytes[4], bytes[5]]);
    |                                             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ expected an array with a fixed size of 4 elements, found one with 8 elements

error: aborting due to previous error

For more information about this error, try `rustc --explain E0308`.
error: could not compile `qp2p`

To learn more, run the command again with --verbose.
warning: build failed, waiting for other jobs to finish...
error: build failed
cargo build --target=i686-unknown-linux-gnu  1171,25s user 83,79s system 143% cpu 14:35,91 total
```

**To Reproduce**
I was trying to compile a Rust FFI library. My `lib.rs`:
```rust
use std::os::raw::c_char;
use std::ffi::CString;
use sn_api::Safe;

// Based on this tutorial: https://dev.to/verkkokauppacom/creating-an-ffi-compatible-c-abi-library-in-rust-5dji

fn xorurl_base() -> String {
    let mut safe = Safe::default();
    return safe.xorurl_base.to_string();
}

#[no_mangle]
pub extern "C" fn c_xorurl_base() -> *mut c_char {
    let rust_string: String = xorurl_base();

    // Convert the String into a CString
    let c_string: CString = CString::new(rust_string).expect("Could not convert to CString");

    // Instead of returning the CString, we return a pointer for it.
    return c_string.into_raw();
}
```

**Expected behavior**
I expect compilation without errors.

 - OS: Arch Linux
 - Shell: zsh
 - Rust: `rustc 1.49.0 (e1884a8e3 2020-12-29)`
 - quic-p2p crate version: 0.10.1

**Comment by loziniak** [comment]
If someone could mentor me a bit, I could try to fix this myself. I'm currently starting to learn Rust.

**Comment by loziniak** [comment]
in comment in `wire_msg.rs` there is:
```
/// | version | message length | `usr_msg_flag` | reserved |
/// | 2 bytes |     4 bytes    |    1 byte    | 2 bytes  |
```
and then:
```
 data_len: usize,
```
Which in my opinion is an error, because `usize` is 8 bytes on 64-bit systems.

**Comment by loziniak** [comment]
Wow, just now I've noticed #213 :-)

**Comment by lionel1704** [comment]
Resolved by #213
